org.osgi.test.common/src/main/java/org/osgi/test/common/event/EventPredicates.java

```diff
@@ -0,0 +1,362 @@
+	}
+
+	static <E> Predicate<List<E>> none(final Predicate<E> predicate) {
+		return elements -> elements.stream()
+			.noneMatch(predicate);
+	}
+
+	@SafeVarargs
+	static <E> Predicate<List<E>> ordered(final Predicate<E> predicate, final Predicate<E>... predicatesNext) {
+		return elements -> {
+
+			List<Predicate<E>> predicates = new ArrayList<>();
+			predicates.add(predicate);
+			predicates.addAll(Arrays.asList(predicatesNext));
+
+			long skip = 0;
+			for (int i = 0; i <= predicatesNext.length - 1; i++) {
+				Predicate<E> predicateBefore = predicates.get(i);
+				Predicate<E> predicateAfter = predicates.get(i + 1);
+
+				Optional<E> before = elements.stream()
+					.skip(skip)
+					.filter(predicateBefore)
+					.findFirst();
+				Optional<E> after = elements.stream()
+					.skip(skip)
+					.filter(predicateAfter)
+					.findFirst();
+
+				if (before.isPresent() && after.isPresent()) {
+
+					int indexBefore = elements.indexOf(before.get());
+					int indexAfter = elements.indexOf(after.get());
+					if (indexBefore >= indexAfter) {
+						return false;
+					}
+					skip = indexBefore;
+				} else {
+					return false;
+				}
+			}
+			return true;
+		};
+	}
+
+	static <E> Predicate<List<E>> hasSize(long size, final Predicate<E> predicate) {
+		return elements -> elements.stream()
+			.filter(predicate)
+			.count() == size;
+	}
+
+	static <E> Predicate<List<E>> hasMoreThen(long size, final Predicate<E> predicate) {
+		return elements -> elements.stream()
+			.filter(predicate)
+			.count() > size;
+	}
+
+	static <E> Predicate<List<E>> hasLessThen(long size, final Predicate<E> predicate) {
+		return elements -> elements.stream()
+			.filter(predicate)
+			.count() < size;
+	}
+
+	class FrameworkEvents {
+
+		// FramworkEvents
+
+		static Predicate<Object> isFrameworkEvent() {
+			return e -> e instanceof FrameworkEvent;
+		}
+
+		static Predicate<Object> isFrameworkEventAnd(Predicate<FrameworkEvent> predicate) {
+			return e -> isFrameworkEvent().test(e) && predicate.test((FrameworkEvent) e);
+		}
+
+		static Predicate<FrameworkEvent> isType(final int eventTypeMask) {
+			return e -> (e.getType() & eventTypeMask) != 0;
+		}
+
+		static Predicate<FrameworkEvent> isTypeError() {
+			return e -> isType(FrameworkEvent.ERROR).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeInfo() {
+			return e -> isType(FrameworkEvent.INFO).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypePackagesRefreshed() {
+			return e -> isType(FrameworkEvent.PACKAGES_REFRESHED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStarted() {
+			return e -> isType(FrameworkEvent.STARTED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStartlevelChanged() {
+			return e -> isType(FrameworkEvent.STARTLEVEL_CHANGED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStopped() {
+			return e -> isType(FrameworkEvent.STOPPED).test(e);
+		}
+
+		static Predicate<FrameworkEvent> isTypeStoppedBootclasspathModified() {
```

Comment:
Don't include this. It was deprecated in Core R8.

Reply:
removed